### ISSUE TITLE
stdlib: Minimum supported versions in VERSIONS file appear to be wrong

### ISSUE THREAD

**Issue by erictraut** [python/typeshed#5328]
I'm adding support for the stdlib VERSIONS file in pyright, and I noticed that a number of modules that I would expect to be available as far back as 3.5 are not. The most prominent one is the `typing` module, which has a minimum required version of Python 3.6, even though it's documented as being available in 3.5 (see PEP 484). There are many other modules that have a minimum version of 3.6 that I suspect are also available in versions older than 3.6.

**Comment by JelleZijlstra** [comment]
I believe this is because typeshed doesn't support versions before 3.6, so we just set 3.6 as the minimum version for all Python 3 libraries. It's harmless to set the precise version for earlier libraries though, so we can just make the changes. Does pyright still actively support 3.5?

**Comment by erictraut** [comment]
Yes, pyright supports 3.0 and newer. I know we have users who maintain code that needs to run in Python 3.5, and it's likely that some users still configure it pyright to assume older versions for compatibility reasons.

**Comment by JelleZijlstra** [comment]
I'm fixing some of these now. A lot of modules marked as new in 3.6 actually already existed in Python 2; the script to generate it must have been buggy.

**Comment by Akuli** [comment]
Is this still an issue? #5329 was merged recently.

**Comment by JelleZijlstra** [comment]
There's a couple of underscored modules that may still be wrong; they're harder to fix because they don't tend to be documented. We can fix those later as needed though.

**Comment by srittau** [comment]
Closing per discussion.
